internal/pkg/manifest/env.go

```diff
@@ -119,11 +120,11 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID                  *string              `yaml:"id,omitempty"`
-	CIDR                *IPNet               `yaml:"cidr,omitempty"`
-	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
-	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
-	Flowlogs            *bool                `yaml:"flow_logs,omitempty"`
+	ID                  *string               `yaml:"id,omitempty"`
+	CIDR                *IPNet                `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration  `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig   `yaml:"security_group,omitempty"`
+	FlowLogs            VPCFlowLogsArgsorBool `yaml:"flow_logs,omitempty"`
```

Comment:
Can we leverage the [`Union` type](https://github.com/aws/copilot-cli/blob/5845a2abfc326641843bb81df286485e2c645665/internal/pkg/manifest/union.go#L14) that @dannyrandall introduced here?

Here are some example usage: https://github.com/aws/copilot-cli/blob/5845a2abfc326641843bb81df286485e2c645665/internal/pkg/manifest/rd_web_svc.go#L86

https://github.com/aws/copilot-cli/blob/5845a2abfc326641843bb81df286485e2c645665/internal/pkg/manifest/svc.go#L376-L386